src/status_im/wallet/choose_recipient/core.cljs

```diff
@@ -75,22 +76,29 @@
      :wallet/keys   [all-tokens] :as db} :db}
    {:keys [chain-id] :as data}
    uri]
-  (let [{:keys [address] :as details}
-        (eip681/extract-request-details data all-tokens)]
-    (if address
-      (if (:wallet/recipient db)
-        {:db (update db :wallet/recipient assoc :resolved-address address
-                     :address address)}
-        (if (:wallet/prepare-transaction db)
-          {:db (update db :wallet/prepare-transaction assoc
-                       :to address :to-name (find-address-name db address))}
-          (let [current-chain-id (get-in networks [current-network :config :NetworkId])]
-            (merge {:db (fill-prepare-transaction-details db details all-tokens)
-                    :dispatch [:open-modal :prepare-send-transaction]}
-                   (when (and chain-id (not= current-chain-id chain-id))
-                     {:ui/show-error (i18n/label :t/wallet-invalid-chain-id
-                                                 {:data uri :chain current-chain-id})})))))
-      {:ui/show-error (i18n/label :t/wallet-invalid-address {:data uri})})))
+  (enc/cond
+    :let [{:keys [address] :as details}
```

Comment:
could you elaborate pls on why `enc/cond`  is used, what is the point? thanks

Reply:
Less nested and more readable code

Reply:
Hi, I'm ok to change this back and use `if` instead. What do you think?

Reply:
it's better to use `if`, because it's more readable if are not familiar with `enc/cond`

Reply:
I just removed usage of `enc/cond` in the latest commit